src/smt/proof_manager.cpp

```diff
@@ -305,5 +340,28 @@ void PfManager::getAssertions(Assertions& as,
   }
 }
 
+void PfManager::getDefinitionsAndAssertions(Assertions& as,
+                                            std::vector<Node>& definitions,
+                                            std::vector<Node>& assertions)
+{
+  for (const Node& d : as.getAssertionListDefinitions())
```

Comment:
Can you store the assertion list definitions into a local before you iterate? I'm not entirely sure how this compile (e.g. if the assertion list is retrieved again on every iteration).

Reply:
hmm... Then I am kinda surprised you didn't complain about line 361 😅